hil/ext/switches/juniper.py

```diff
@@ -0,0 +1,294 @@
+        jun = self.session()
+        try: 
+            with jun:
+                interface_config = InterfaceConfigTables(jun)
+                interface_config.get(interface= port, options = {'database':'committed'})
+                ic_dict = json.loads(interface_config.to_json())
+        except ConnectError as err:
+                print("connat connect to device: {0}".format(err))
+                sys.exit(1)
+
+        return ic_dict
+
+    def _trunk_mode(self, port):
+        """Returns True if the port is in trunk mode else returns False. """
+        port_info = self._interface_info(port)
+        return port_info[port]['trunk_mode']
+
+    def _set_mode(self, port, mode):
+        """ Will set a given port to 
+        trunk mode or access mode based on the `mode` flag.
+        """
+        if mode == "trunk":
+            set_mode = """
+            set interfaces {port} unit 0 family ethernet-switching interface-mode trunk
+        """.format(port=port)
+            self._set_commit_config(set_mode)
+        else: 
+            self._remove_all_vlans_port(port)
+
+        #x = "set interfaces et-0/0/08:0 unit 0 family ethernet-switching interface-mode trunk"
+
+    def _get_native_vlan(self, port):
+        """ Returns the id of the native vlan for the trunked port 
+        if it exists returns nothing otherwise.  
+        """ 
+        port_info = self._interface_info(port)
+        return port_info[port]['native_vlan']
+
+    def _set_native_vlan(self, port, network_id):
+       """Sets native vlan for a trunked port. """
+       set_native_vlan = """
+       set interface {port} native-vlan-id {network_id}
+        """.format(port=port, network_id=network_id)
+       self._set_commit_config(set_native_vlan)
+
+    def _remove_native_vlan(self, port, network_id):
+        """Removes native_vlan tag from port. """
+
+        remove_native_vlan = """
+        delete interface {port} native-vlan-id {network_id}
+        """.format(port=port)
+        self._set_commit_config(remove_native_vlan)
+
+
+                
+    def _add_vlan_to_trunk(self, port, vlan_id):
+        """ Adds vlans to a trunk port. """
+        #eg: set interfaces et-0/0/08:0 unit 0 family ethernet-switching vlan members 200
+
+        add_vlan = """
+        set interfaces {port} unit 0 family ethernet-switching vlan members {vlan_id}
+        """.format(port=port, vlan_id=vlan_id)
+
+        self._set_commit_config(add_vlan)
+
+    def _remove_vlan_from_port(self, port, vlan_id):
+        """ removes a single vlan specified by `vlan_id` """
+        remove_vlan = """
+        delete interfaces {port} unit 0 family ethernet-switching vlan members {vlan_id}
+        """.format(port=port, vlan_id=vlan_id)
+        self._set_commit_config(remove_vlan)
+        
+    def _remove_all_vlans_port(self, port):
+        """ Removes all vlans from the port, including the native vlan.
+        Also converts the interface to access mode. 
```

Comment:
2 questions:

1. If this method can remove all vlans, then why do we need the other revert port? 
2. How/Why does removing all the VLANs puts it into access mode? Is this how the switch works?

Reply:
Looks like I will not be needing this function after all. Removed it.